### DISCUSSION TITLE
measuring current thrust acting along body z-axis

### DISCUSSION THREAD

**Discussion by adarsh2798** [bitcraze/discussions#1328] [Q&A]
Hello,

I need to measure the current thrust acting along body z-axis of crazyflie. I dont need the commanded thrust, but the actual thrust along body z-axis which i can use as a feedback for a closed-loop acceleration control.  
In Logging, motor group retruns PWM and I also found a PWM to thrust mapping, but I think the PWM it logs is the commnaded PWM that is found in the lowest level after receiving any high level control signal.

To find current actual thrust, can I measure propellor speeds?
This link https://www.bitcraze.io/2015/01/measuring-propeller-rpm-part-1/ provides a tutorial for that using QRD1114 photodiode, but I dont know how effectuve it will be subject to ambien light. Also I am a bit confused where exaclty to connect the sensor output to crazyflie's onboard stm microcontroller. If there is a clear schematic regarding the same it would be helpful!

thanks.

**Comment by gemenerik** [comment]
I'll see if we can help you reproduce the deck. Note, however, that estimating dynamic thrust accurately is not trivial. E.g., think of the ground effect, how drone velocity affects it, etc. Depending on what accuracy you are looking for, propeller RPM may not be sufficient.

**Comment by NageshEranki** [comment]
This might be coming a little late but, maybe you can log the accelerometer values along the z-axis? The units are in 'g's if I remember. These readings are the actual specific force along z axis.
